api/v1/controllers/RSVPController.js

```diff
@@ -0,0 +1,152 @@
+}
+
+function createRSVP(req, res, next) {
+    if(!req.body.isAttending)
+        delete req.body.type;
+
+    services.RegistrationService
+        .findAttendeeByUser(req.user)
+        .then(function(attendee) {
+           return services.RSVPService
+                .createRSVP(attendee, req.user, req.body);
+        })
+        .then(function(rsvp) {
+            if(rsvp.get('isAttending'))
+                services.MailService.addToList(req.user, mail.lists.attendees);
+            res.body = rsvp.toJSON();
+
+            return next();
+        })
+        .catch(function(error) {
+            return next(error);
+        });
+}
+
+function fetchRSVPByUser(req, res, next) {
+    services.RegistrationService
+        .findAttendeeByUser(req.user)
+        .then(function(attendee) {
+            return services.RSVPService
+                .findRSVPByAttendee(attendee);
+        })
+        .then(function (rsvp) {
+            res.body = rsvp.toJSON();
+            if(!res.body.type) {
+                delete res.body.type;
+            }
+
+            return next();
+        })
+        .catch(function(error) {
+            return next(error);
+        })
+}
+
+function fetchRSVPByAttendeeId(req, res, next) {
+    services.RegistrationService
+        .findAttendeeById(req.params.id)
+        .then(function(attendee) {
+            services.RSVPService
+                .findRSVPByAttendee(attendee);
+        })
+        .then(function(rsvp){
+            res.body = rsvp.toJSON();
+            if(!res.body.type) {
+                delete res.body.type;
+            }
+
+            return next();
+        })
+        .catch(function(error) {
+            return next(error);
+        })
+}
+
+function updateRSVPByUser(req, res, next) {
+    if(!req.body.isAttending)
+        delete req.body.type;
+
+    services.RegistrationService
+        .findAttendeeByUser(req.user)
+        .then(function(attendee) {
+            return _updateRSVPByAttendee(req.user, attendee, req.body);
+        })
+        .then(function(rsvp){
+            res.body = rsvp.toJSON();
+
+            return next();
+        })
+        .catch(function (error) {
+            return next(error);
+        });
+}
+
+function updateRSVPById(req, res, next) {
+    if(!req.body.isAttending)
+        delete req.body.type;
+
+    services.RegistrationService
+        .findAttendeeById(req.params.id)
+        .then(function(attendee) {
+            return _updateRSVPByAttendee(req.user, attendee, req.body)
+        })
+        .then(function(rsvp){
+            res.body = rsvp.toJSON();
+
+            return next();
+        })
+        .catch(function (error) {
+            return next(error);
+        });
+}
+
+function _updateRSVPByAttendee(user, attendee, newRSVP) {
+    return services.RSVPService
+        .findRSVPByAttendee(attendee)
+        .then(function (rsvp) {
+            if(_addToList(rsvp, newRSVP))
```

Comment:
Add/remove people to/from lists after the update is complete. If you do it before, and error might occur later and they might be stuck on the wrong list.